src/main/java/com/gmail/goosius/siegewar/settings/SiegeWarSettings.java

```diff
@@ -763,19 +766,42 @@ private static void addSpecialEffects(ItemStack artefact, ItemMeta itemMeta, Str
         }
 		//Add special effects
 		for(String[] effectSpec: effectSpecs) {
+		    if (!validateEffect(material, effectSpec[0]))
+		        return false;
 			addSpecialEffect(material, itemMeta, effectSpec);
 		}
 		//Set updated item meta
 		artefact.setItemMeta(itemMeta);
+		return true;
 	}
 
-	private static void addSpecialEffect(Material material, ItemMeta itemMeta, String[] effectSpec) {
-		if(effectSpec[0].equalsIgnoreCase("custom_effect")) {
-			addCustomEffect(itemMeta, effectSpec);
-		} else if(material == Material.POTION
+	// Make sure that the config has been given a valid PotionEffectType or Enchantment name.
+    private static boolean validateEffect(Material material, String name) {
+        return !name.isEmpty()
+                && name.equals("custom_effect") // ignore custom_effect
+                || ((isPotionBased(material) && !validatePotionEffectType(name))
+                    || !isPotionBased(material) && !validateEnchantmentType(name));

```

Comment:
🚧 
It looks like...

* !validatePotionEffectType... should not have the exclamation mark.
* !validateEnchantmentType.... should also not have the exclamation mark.

Also I just can't tell what the top level logic will do. If you have any uncertainty also I suggest:
```
return !name.isEmpty()
             &&
            (
                  name.equals("custom_effect") // ignore custom_effect
                   || 
                   (isPotionBased(material).... )
                   || 
                   (!isPotionBased(material)...)
             );
```